// next-image-export-optimizer/dist/ExportedImage.js
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
exports.default = void 0;
var _dynamic = _interopRequireDefault(require("next/dynamic"));
var _image = _interopRequireDefault(require("next/image"));
var _react = _interopRequireWildcard(require("react"));
function _arrayLikeToArray(arr, len) {
    if (len == null || len > arr.length) len = arr.length;
    for(var i = 0, arr2 = new Array(len); i < len; i++)arr2[i] = arr[i];
    return arr2;
}
function _arrayWithHoles(arr) {
    if (Array.isArray(arr)) return arr;
}
function _extends() {
    _extends = Object.assign || function(target) {
        for(var i = 1; i < arguments.length; i++){
            var source = arguments[i];
            for(var key in source){
                if (Object.prototype.hasOwnProperty.call(source, key)) {
                    target[key] = source[key];
                }
            }
        }
        return target;
    };
    return _extends.apply(this, arguments);
}
function _interopRequireDefault(obj) {
    return obj && obj.__esModule ? obj : {
        default: obj
    };
}
function _getRequireWildcardCache() {
    if (typeof WeakMap !== "function") return null;
    var cache = new WeakMap();
    _getRequireWildcardCache = function() {
        return cache;
    };
    return cache;
}
function _interopRequireWildcard(obj) {
    if (obj && obj.__esModule) {
        return obj;
    }
    if (obj === null || typeof obj !== "object" && typeof obj !== "function") {
        return {
            default: obj
        };
    }
    var cache = _getRequireWildcardCache();
    if (cache && cache.has(obj)) {
        return cache.get(obj);
    }
    var newObj = {};
    var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor;
    for(var key in obj){
        if (Object.prototype.hasOwnProperty.call(obj, key)) {
            var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null;
            if (desc && (desc.get || desc.set)) {
                Object.defineProperty(newObj, key, desc);
            } else {
                newObj[key] = obj[key];
            }
        }
    }
    newObj.default = obj;
    if (cache) {
        cache.set(obj, newObj);
    }
    return newObj;
}
function _iterableToArrayLimit(arr, i) {
    var _i = arr == null ? null : typeof Symbol !== "undefined" && arr[Symbol.iterator] || arr["@@iterator"];
    if (_i == null) return;
    var _arr = [];
    var _n = true;
    var _d = false;
    var _s, _e;
    try {
        for(_i = _i.call(arr); !(_n = (_s = _i.next()).done); _n = true){
            _arr.push(_s.value);
            if (i && _arr.length === i) break;
        }
    } catch (err) {
        _d = true;
        _e = err;
    } finally{
        try {
            if (!_n && _i["return"] != null) _i["return"]();
        } finally{
            if (_d) throw _e;
        }
    }
    return _arr;
}
function _nonIterableRest() {
    throw new TypeError("Invalid attempt to destructure non-iterable instance.\\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.");
}
function _objectWithoutProperties(source, excluded) {
    if (source == null) return {};
    var target = _objectWithoutPropertiesLoose(source, excluded);
    var key, i;
    if (Object.getOwnPropertySymbols) {
        var sourceSymbolKeys = Object.getOwnPropertySymbols(source);
        for(i = 0; i < sourceSymbolKeys.length; i++){
            key = sourceSymbolKeys[i];
            if (excluded.indexOf(key) >= 0) continue;
            if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue;
            target[key] = source[key];
        }
    }
    return target;
}
function _objectWithoutPropertiesLoose(source, excluded) {
    if (source == null) return {};
    var target = {};
    var sourceKeys = Object.keys(source);
    var key, i;
    for(i = 0; i < sourceKeys.length; i++){
        key = sourceKeys[i];
        if (excluded.indexOf(key) >= 0) continue;
        target[key] = source[key];
    }
    return target;
}
function _slicedToArray(arr, i) {
    return _arrayWithHoles(arr) || _iterableToArrayLimit(arr, i) || _unsupportedIterableToArray(arr, i) || _nonIterableRest();
}
function _unsupportedIterableToArray(o, minLen) {
    if (!o) return;
    if (typeof o === "string") return _arrayLikeToArray(o, minLen);
    var n = Object.prototype.toString.call(o).slice(8, -1);
    if (n === "Object" && o.constructor) n = o.constructor.name;
    if (n === "Map" || n === "Set") return Array.from(n);
    if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen);
}
var splitFilePath = function(param) {
    var filePath = param.filePath;
    var ref;
    var filenameWithExtension = ((ref = filePath.split("\\").pop()) === null || ref === void 0 ? void 0 : ref.split("/").pop()) || "";
    var filePathWithoutFilename = filePath.split(filenameWithExtension).shift();
    var fileExtension = filePath.split(".").pop();
    var filenameWithoutExtension = filenameWithExtension.substring(0, filenameWithExtension.lastIndexOf(".")) || filenameWithExtension;
    return {
        path: filePathWithoutFilename,
        filename: filenameWithoutExtension,
        extension: fileExtension || ""
    };
};
var generateImageURL = function(src, width) {
    var ref = splitFilePath({
        filePath: src
    }), filename = ref.filename, path = ref.path, extension = ref.extension;
    if (![
        "JPG",
        "JPEG",
        "WEBP",
        "PNG",
        "AVIF"
    ].includes(extension.toUpperCase())) {
        return src;
    }
    var processedExtension = extension;
    if (process.env.storePicturesInWEBP === true && [
        "JPG",
        "JPEG",
        "PNG"
    ].includes(extension.toUpperCase())) {
        processedExtension = "WEBP";
    }
    var correctedPath = path;
    var lastChar = correctedPath === null || correctedPath === void 0 ? void 0 : correctedPath.substr(-1);
    if (lastChar != "/") {
        correctedPath = correctedPath + "/";
    }
    return "".concat(correctedPath, "nextImageExportOptimizer/").concat(filename, "-opt-").concat(width, ".").concat(processedExtension.toUpperCase());
};
var optimizedLoader = function(param) {
    var src = param.src, width = param.width;
    return generateImageURL(src, width);
};
var fallbackLoader = function(param) {
    var src = param.src;
    return src;
};
function ExportedImage(_param) {
    var src = _param.src, _priority = _param.priority, priority = _priority === void 0 ? false : _priority, loading = _param.loading, _lazyRoot = _param.lazyRoot, lazyRoot = _lazyRoot === void 0 ? null : _lazyRoot, _lazyBoundary = _param.lazyBoundary, lazyBoundary = _lazyBoundary === void 0 ? "200px" : _lazyBoundary, className = _param.className, quality = _param.quality, width = _param.width, height = _param.height, objectFit = _param.objectFit, objectPosition = _param.objectPosition, onLoadingComplete = _param.onLoadingComplete, _placeholder = _param.placeholder, placeholder = _placeholder === void 0 ? process.env.generateAndUseBlurImages === true ? "blur" : "empty" : _placeholder, blurDataURL = _param.blurDataURL, rest = _objectWithoutProperties(_param, [
        "src",
        "priority",
        "loading",
        "lazyRoot",
        "lazyBoundary",
        "className",
        "quality",
        "width",
        "height",
        "objectFit",
        "objectPosition",
        "onLoadingComplete",
        "placeholder",
        "blurDataURL"
    ]);
    var ref = _slicedToArray((0, _react).useState(false), 2), imageError = ref[0], setImageError = ref[1];
    var automaticallyCalculatedBlurDataURL = (0, _react).useMemo(function() {
        if (blurDataURL) {
            return blurDataURL;
        }
        return generateImageURL(src, 10);
    }, [
        blurDataURL,
        src
    ]);
    return _react.default.createElement(_image.default, _extends({}, rest, width && {
        width: width
    }, height && {
        height: height
    }, priority && {
        priority: priority
    }, loading && {
        loading: loading
    }, lazyRoot && {
        lazyRoot: lazyRoot
    }, lazyBoundary && {
        lazyBoundary: lazyBoundary
    }, className && {
        className: className
    }, quality && {
        quality: quality
    }, objectFit && {
        objectFit: objectFit
    }, objectPosition && {
        objectPosition: objectPosition
    }, onLoadingComplete && {
        onLoadingComplete: onLoadingComplete
    }, placeholder && {
        placeholder: placeholder
    }, {
        loader: imageError ? fallbackLoader : optimizedLoader,
        blurDataURL: automaticallyCalculatedBlurDataURL,
        src: src,
        onError: function() {
            setImageError(true);
        }
    }));
}
var _default = (0, _dynamic).default(function() {
    return Promise.resolve(ExportedImage);
}, {
    ssr: false
});
exports.default = _default;
